quadtree.js

```diff
@@ -37,21 +37,31 @@ class Rectangle {
   }
 
   contains(point) {
-    return (point.x >= this.x - this.w &&
-      point.x <= this.x + this.w &&
-      point.y >= this.y - this.h &&
-      point.y <= this.y + this.h);
+    return (
+      this.left <= point.x && point.x < this.right &&
+      this.top <= point.y && point.y < this.bottom
+    );
```

Comment:
This is much better, using `this.left` and the like. I'm wondering if maybe we should change the edges from functions to saved attributes set in the constructor? Might save a little math.

Reply:
```JS
constructor(x, y, w, h) {
    this.x = x;
    this.y = y;
    this.w = w;
    this.h = h;
    this.left = x - w / 2;
    // etc.
  }
```

Reply:
Ah yeah, can do. Calculate them once rather than each time. 👍🏻 

Sounds good.

Reply:
Also... there's discussion to be had around how the `contains` works. It's used in two places and I think it should work different for each.

First case is when adding a point and finding which quadrant to add it to. In this case my new logic should be used.

So that if the point lands on the border of `ne` and `nw` (for example) then it will always be put into `nw` and never put into `ne`.

But for querying all points in a rect I think it should use the `<=` on both sides so that it doesn't miss out any (literal) edge cases.

In which case... we would need two separate functions to do these.